### PR TITLE
removes -mod=vendor flag from build command

### LINKED ISSUE #66: Can remove `-mod=vendor` flag logic
[As of Go 1.14](https://golang.org/doc/go1.14#go-command) there is no longer a need to set the `-mod=vendor` explicitly.

Therefore the following logic can be removed https://github.com/paketo-buildpacks/go-build/blob/7c0f13016c3f789bea90e7f91a2a925fdcac9a88/go_build_process.go#L68-L74

### PR DESCRIPTION
**Pull request by fg-j**
Resolves #66

### CODE REVIEW

**Comment by ForestEckhardt** [comment]
@fg-j I think that the PR look good. I think that there are a couple of integration tests that can be removed now because of this change. Particularly, integration tests have using the `mod` test app are no longer relevant. @ryanmoran are there any other tests that you could think of as being able to be removed?

**Comment by ForestEckhardt** [comment]
I had a quick conversation with @ryanmoran, the only test we could see that should be removed is any test that uses the `mod` app.

File: integration/mod_test.go [hunk 1]
```diff
@@ -1,157 +0,0 @@
...
-				"  Executing build process",
-				fmt.Sprintf("    Running 'go build -o /layers/%s/targets/bin -buildmode pie -mod vendor .'", strings.ReplaceAll(settings.Buildpack.ID, "/", "_")),
-				MatchRegexp(`      Completed in ([0-9]*(\.[0-9]*)?[a-z]+)+`),
-				"",
-				"  Assigning launch processes",
-				fmt.Sprintf("    web: /layers/%s/targets/bin/mod_vendor", strings.ReplaceAll(settings.Buildpack.ID, "/", "_")),
-			))
-		})
-	})
-}
```
Comment (ForestEckhardt) [inline integration/mod_test.go:157]: I think that we actually probably still want this test but I don't know where it should go. Probably in a new test file.

Comment (fg-j) [reply integration/mod_test.go:157]: Actually, I'm having a hard time seeing the difference between the above test (in the case where the -mod vendor flag is used) and [this test](https://github.com/fg-j/go-build/blob/872fbc4c93b9daee6f0e19d27f51f8e0913b7dc3/integration/vendor_test.go#L95).

Comment (fg-j) [reply integration/mod_test.go:157]: The release notes linked in #66 state:

> When the main module contains a top-level vendor directory and its go.mod file specifies go 1.14 or higher, the go command now defaults to -mod=vendor for operations that accept that flag. A new value for that flag, -mod=mod, causes the go command to instead load modules from the module cache (as when no vendor directory is present).

Not clear from that what the expected behaviour is in the absence of a `go.mod`.

Comment (ryanmoran) [reply integration/mod_test.go:157]: We still have code that we want to test that relies on the presence of a `go.mod` file: https://github.com/paketo-buildpacks/go-build/blob/f9f97ca97cfc3ecbbe859ac4843bdf88c30e6c70/go_path_manager.go#L23-L26.

I think we can remove the vendored mod test and its fixture, but keep the non-vendored mod test.

**Review by ryanmoran — CHANGES_REQUESTED** [review]

**Review by ryanmoran — APPROVED** [review]

**Pull request merged**